src/metabase/sync/util.clj

```diff
@@ -458,10 +458,7 @@
                   :task_details (when (seq task-details)
                                   task-details)))
          (cons (create-task-history operation database sync-md))
-         ;; Using `insert!` instead of `insert-many!` here to make sure
-         ;; `post-insert` is triggered
-         (map #(db/insert! TaskHistory %))
-         (map :id)
+         (map #(t2/insert-returning-pks! TaskHistory %))
```

Comment:
```suggestion
         (t2/insert-returning-pks! TaskHistory)
```

(And there's definitely a bug as written right now, I think it'll return a nested list of IDs: `((1) (2) (3))` instead of `(1 2 3)`)

Reply:
this is a blocker